rosidl_typesupport_c/QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,99 @@
+This document is a declaration of software quality for the `rosidl_typesupport_c` package, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+# `rosidl_typesupport_c` Quality Declaration
+
+The package `rosidl_typesupport_c` claims to be in the **Quality Level 4** category.
+
+Below are the rationales, notes, and caveats for this claim, organized by each requirement listed in the [Package Requirements for Quality Level 4 in REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+## Version Policy
```

Comment:
In the REP 2004 there is a guideline for Quality Declaration in `markdown` which add the index referencing the REP to the title. Right now looks like:

```
## Version Policy
```
But the guideline define this as
```
## Version Policy [1]
```

I suggest to change all the titles to follow the section number as defined in the REP

Reply:
Titles have been updated accordingly.